`TestTeleportProcessAuthVersionCheck` flakiness

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/3808447821/jobs/6479033908
- https://github.com/gravitational/teleport/actions/runs/5059284810/jobs/9080594192#step:6:326

#### Relevant snippet

```
service_test.go:691: 
[279](https://github.com/gravitational/teleport/actions/runs/3808447821/jobs/6479033908#step:6:280)
        	Error Trace:	/__w/teleport/teleport/lib/service/service_test.go:691
[280](https://github.com/gravitational/teleport/actions/runs/3808447821/jobs/6479033908#step:6:281)
        	Error:      	Received unexpected error:
[281](https://github.com/gravitational/teleport/actions/runs/3808447821/jobs/6479033908#step:6:282)
        	            	listen tcp 127.0.0.1:44731: bind: address already in use
[282](https://github.com/gravitational/teleport/actions/runs/3808447821/jobs/6479033908#step:6:283)
        	Test:       	TestTeleportProcessAuthVersionCheck
[283](https://github.com/gravitational/teleport/actions/runs/3808447821/jobs/6479033908#step:6:284)
--- FAIL: TestTeleportProcessAuthVersionCheck (1.47s)
```

--- a/lib/service/service_test.go
+++ b/lib/service/service_test.go
@@ -865,26 +865,8 @@ func TestTeleportProcessAuthVersionCheck(t *testing.T) {
 	lib.SetInsecureDevMode(true)
 	defer lib.SetInsecureDevMode(false)
 
-	authAddr, err := getFreePort()
-	require.NoError(t, err)
-	listenAddr := utils.NetAddr{AddrNetwork: "tcp", Addr: authAddr}
+	listenAddr := utils.NetAddr{AddrNetwork: "tcp", Addr: "127.0.0.1:0"}
 	token := "join-token"
-
-	// Create Node process.
-	nodeCfg := servicecfg.MakeDefaultConfig()
-	nodeCfg.SetAuthServerAddress(listenAddr)
-	nodeCfg.DataDir = t.TempDir()
-	nodeCfg.SetToken(token)
-	nodeCfg.Auth.Enabled = false
-	nodeCfg.Proxy.Enabled = false
-	nodeCfg.SSH.Enabled = true
-
-	// Set the Node's major version to be greater than the Auth Service's,
-	// which should make the version check fail.
-	currentVersion, err := semver.NewVersion(teleport.Version)
-	require.NoError(t, err)
-	currentVersion.Major++
-	nodeCfg.TeleportVersion = currentVersion.String()
 
 	// Create Auth Service process.
 	staticTokens, err := types.NewStaticTokens(types.StaticTokensSpecV2{
@@ -919,6 +901,23 @@ func TestTeleportProcessAuthVersionCheck(t *testing.T) {
 		authProc.Close()
 	})
 
+	// Create Node process, pointing at the auth server's local port
+	authListenAddr := authProc.Config.AuthServerAddresses()[0]
+	nodeCfg := servicecfg.MakeDefaultConfig()
+	nodeCfg.SetAuthServerAddress(authListenAddr)
+	nodeCfg.DataDir = t.TempDir()
+	nodeCfg.SetToken(token)
+	nodeCfg.Auth.Enabled = false
+	nodeCfg.Proxy.Enabled = false
+	nodeCfg.SSH.Enabled = true
+
+	// Set the Node's major version to be greater than the Auth Service's,
+	// which should make the version check fail.
+	currentVersion, err := semver.NewVersion(teleport.Version)
+	require.NoError(t, err)
+	currentVersion.Major++
+	nodeCfg.TeleportVersion = currentVersion.String()
+
 	t.Run("with version check", func(t *testing.T) {
 		testVersionCheck(t, nodeCfg, false)
 	})
@@ -946,20 +945,6 @@ func testVersionCheck(t *testing.T, nodeCfg *servicecfg.Config, skipVersionCheck
 	supervisor, ok := nodeProc.Supervisor.(*LocalSupervisor)
 	require.True(t, ok)
 	supervisor.signalExit()
-}
-
-func getFreePort() (string, error) {
-	addr, err := net.ResolveTCPAddr("tcp", "127.0.0.1:0")
-	if err != nil {
-		return "", err
-	}
-	l, err := net.ListenTCP("tcp", addr)
-	if err != nil {
-		return "", err
-	}
-	defer l.Close()
-
-	return l.Addr().(*net.TCPAddr).String(), nil
 }
 
 func Test_readOrGenerateHostID(t *testing.T) {

